chia/full_node/full_node.py

```diff
@@ -1372,6 +1382,37 @@ async def peak_post_processing_2(
         await self.server.send_to_all([msg], NodeType.WALLET)
         self._state_changed("new_peak")
 
+        # Update certain prometheus values that we can't update with a known value in real time elsewhere
+        # Skip updating these values (which result in a few DB queries) if the prometheus server is not enabled
+        if self.prometheus.server.server_enabled:
```

Comment:
1. If you don't create the instance at all, then you're dealing with a bunch of conditional logic all over the place on whether or not you should even call the prometheus methods, and it clutters up the full node, etc code with tons of duplicated conditionals - seemed easier to deal with that internal to prometheus, and not make the code outside of that (usually) care whether or not the server is started or not, and just call the update metric methods either way

Exception to that was this section here, where these metrics values actually had to make DB queries to set the values properly, since we don't otherwise have things like # of compact blocks, etc handy. The reason this was referencing server_enabled external to the Prom class was because we needed the block store, but we could probably just pass the block store to the prometheus class, and just call a method on that here, that will do this logic internal to the class, and avoids checking this conditional outside, which is more inline with what I outlined above as my goal (making the code external to prometheus not really care about the internal state of prometheus - started or not - and just passing new metric values when it has them)

Definitely agree that we don't want any metrics errors to take down the full node, will look at your `with self.prometheus.server.log_errors()` suggestion tomorrow

Reply:
All exceptions are caught with `with self.prometheus.server.log_errors():` 

I'm no longer checking `server_enabled` outside of the class - instead, I just check it immediately upon entering the metrics update methods on the class, and if its disabled, that whole process is just skipped and no metrics are set/updated. 

Currently, the config for prometheus (enabled, port number, etc) is checked inside of the prometheus class, so not setting the Prom object at all would require moving config logic external to the prom class - thinking it might just make sense to leave it this way, since all the methods/server starting/etc is all short circuited if its not enabled.

Also renamed the start method to start_if_enabled, to make it more clear what is going on there.